src/services/job_service.py

```diff
@@ -96,25 +112,58 @@ def _get_job(self, job_uid: UUID) -> Union[Job, None]:
 
     @staticmethod
     def _get_job_entity(dmss_id: str, token: str = None):
+        """Get a document from DMSS.
+
+        The dmss_id can be on the formats:
+          - By id: PROTOCOL://DATA SOURCE/$ID.Attribute
+          - By path: PROTOCOL://DATA SOURCE/ROOT PACKAGE/SUB PACKAGE/ENTITY.Attribute
+        """
         data_source_id, job_entity_id, attribute = split_absolute_ref(dmss_id)
         return get_document_by_uid(
             reference=f"{data_source_id}/{job_entity_id}.{attribute}", token=token, depth=50, resolve_links=False
         )
 
     @staticmethod
     def _insert_reference(document_id: str, reference: dict, token: str = ""):  # nosec
+        """Insert a reference into an existing entity stored in dmss.
+
+        - **document_id**: the address to the entity we want to update. Can be on the formats:
+          - By id: PROTOCOL://DATA SOURCE/$ID.Attribute
+          - By path: PROTOCOL://DATA SOURCE/ROOT PACKAGE/SUB PACKAGE/ENTITY.Attribute
+
+        - **reference**: an entity of type 'dmss://system/SIMOS/Reference' to be inserted.
+        """
         headers = {"Access-Key": token}
+        # TODO use document update instead, the reference insert endpoint has been removed from DMSS
         req = requests.put(f"{config.DMSS_API}/api/reference/{document_id}", json=reference, headers=headers)
```

Comment:
Didn't you delete this endpoint?

Reply:
yes, that's why the TODO is added. Also added an issue for it. https://github.com/equinor/dm-job/issues/42

Reply:
Haha, I didn't notice